### DISCUSSION TITLE
Handling small gap and holes, high-performance merging methods

### DISCUSSION THREAD

**Discussion by madeer** [locationtech/jts#1068] [Q&A]
We have an actual business scenario like this. Employees will draw a lot of two-dimensional polygons or multipolygons, but the drawings they draw are not standard.there are very small gaps or holes between two adjacent polygons . This causes two problems. One is that the merging of many geometries is slow. There is no more performant merge algorithm. Another problem is that there are many useless gaps and small holes after merging. Is there any good solution? I have been looking for a long time but I have not found a good solution. I look forward to your reply! The merging method I use now is to convert all geometries into one geometry first, and then buffer it, but this will also cause many problems, and I will not discuss the specific issues for the time being.![IMG_20240816_233215.png](https://github.com/user-attachments/assets/876c8f66-5ec1-41ec-bdf8-e28d0bd77709)

**Comment by micycle1** [comment]
Try `org.locationtech.jts.operation.union.UnaryUnionOp.union()`.

>Unions a Collection of Geometrys or a single Geometry(which may be a GeometryCollection) together. By using this special-purpose operation over a collection of geometries it is possible to take advantage of various optimizations to improve performance. Heterogeneous GeometryCollections are fully supported.

**Comment by dr-jts** [comment]
Really you need better tools to provide some snapping when polygons are created.

Otherwise, you can remove narrow holes after unioning.  Currently there's no JTS operations to do snapping effectively/easily on polygonal coverages.  Perhaps there will be someday.
